apd.Decimal is decoded incorrectly

```
                testdata/logic_test/window:212: SELECT *, avg(k) OVER (w ORDER BY w) FROM kv WINDOW w AS (PARTITION BY v) ORDER BY 1
                expected:
                    k  v     w  f    d     s     b      i             avg
                    1  2     3  1    1     a     true   00:01:00      4.6666666666666666667
                    3  4     5  2    8     a     true   00:00:02      5.5
                    5  NULL  5  9.9  -321  NULL  false  NULL          5
                    6  2     3  4.4  4.4   b     true   00:00:00.001  4.6666666666666666667
                    7  2     2  6    7.9   b     true   4 days        7
                    8  4     2  3    3     A     false  3 years       8

                but found (query options: "colnames") :
                    k  v     w  f    d     s     b      i             avg
                    1  2     3  1    1     a     true   00:01:00      4.6666666666666666667
                    3  4     5  2    8     a     true   00:00:02      5.5
                    5  NULL  5  9.9  -321  NULL  false  NULL          5
                    6  2     3  4.4  -4.4  b     true   00:00:00.001  4.6666666666666666667
                    7  2     2  6    7.9   b     true   4 days        7
                    8  4     2  3    3     A     false  3 years       8
```

Parent issue: #46185

Reprod with `local-vec` config.

kv trace: 
```
    --- FAIL: TestExecBuild/local-vec (0.33s)
        --- FAIL: TestExecBuild/local-vec/window (0.32s)
            logic.go:1158: overriding vectorize mode to on, setting batch size to 1
            logic.go:1835:

                testdata/window:29: SELECT message FROM [SHOW KV TRACE FOR SESSION] WITH ORDINALITY
                 WHERE message LIKE 'fetched:%' OR message LIKE 'output row%'
                 ORDER BY message LIKE 'fetched:%' DESC, ordinality ASC
                expected:
                    fetched: /kv/primary/1/v -> /2
                    fetched: /kv/primary/1/d -> 1
                    fetched: /kv/primary/1/s -> 'a'
                    fetched: /kv/primary/3/v -> /4
                    fetched: /kv/primary/3/d -> 8
                    fetched: /kv/primary/3/s -> 'a'
                    fetched: /kv/primary/5 -> NULL
                    fetched: /kv/primary/5/d -> -321
                    fetched: /kv/primary/6/v -> /2
                    fetched: /kv/primary/6/d -> 4.4
                    fetched: /kv/primary/6/s -> 'b'
                    fetched: /kv/primary/7/v -> /2
                    fetched: /kv/primary/7/d -> 7.9
                    fetched: /kv/primary/7/s -> 'b'
                    fetched: /kv/primary/8/v -> /4
                    fetched: /kv/primary/8/d -> 3
                    fetched: /kv/primary/8/s -> 'A'
                    output row: [5 NULL]
                    output row: [1 3.4501207708330056852]
                    output row: [6 3.4501207708330056852]
                    output row: [7 3.4501207708330056852]
                    output row: [3 3.5355339059327376220]
                    output row: [8 3.5355339059327376220]

                but found (query options: "rowsort" -> ignore the following ordering of rows) :
                    fetched: /kv/primary/1/v -> /2
                    fetched: /kv/primary/1/d -> 1
                    fetched: /kv/primary/1/s -> 'a'
                    fetched: /kv/primary/3/v -> /4
                    fetched: /kv/primary/3/d -> 8
                    fetched: /kv/primary/3/s -> 'a'
                    fetched: /kv/primary/5 -> NULL
                    fetched: /kv/primary/5/d -> -321
                    fetched: /kv/primary/6/v -> /2
                    fetched: /kv/primary/6/d -> -4.4
                    fetched: /kv/primary/6/s -> 'b'
                    fetched: /kv/primary/7/v -> /2
                    fetched: /kv/primary/7/d -> -7.9
                    fetched: /kv/primary/7/s -> 'b'
                    fetched: /kv/primary/8/v -> /4
                    fetched: /kv/primary/8/d -> -3
                    fetched: /kv/primary/8/s -> 'A'
                    output row: [5 NULL]
                    output row: [1 4.4836740886613662671]
                    output row: [6 4.4836740886613662671]
                    output row: [7 4.4836740886613662671]
                    output row: [3 7.7781745930520227684]
                    output row: [8 7.7781745930520227684]
```

Query Plan
```
  │
  └ Node 1
    └ *colexec.Materializer
      └ *colexec.diskSpillerBase
        ├ *colexec.sortOp
        │ └ *colexec.allSpooler
        │   └ *colexec.Columnarizer
        │     └ *rowexec.windower
        │       └ *colexec.Materializer
        │         └ *colexec.CancelChecker
        │           └ *colexec.colBatchScan
        ├ *colexec.Columnarizer
        └ *colexec.externalSorter
          └ *colexec.sortOp
            └ *colexec.allSpooler
              └ *colexec.inputPartitioningOperator
                └ *colexec.bufferExportingOperator
```

--- a/pkg/util/encoding/decimal.go
+++ b/pkg/util/encoding/decimal.go
@@ -624,6 +624,7 @@ func DecodeNonsortingDecimal(buf []byte, tmp []byte) (apd.Decimal, error) {
 // DecodeIntoNonsortingDecimal is like DecodeNonsortingDecimal, but it operates
 // on the passed-in *apd.Decimal instead of producing a new one.
 func DecodeIntoNonsortingDecimal(dec *apd.Decimal, buf []byte, tmp []byte) error {
+	*dec = apd.Decimal{}
 	switch buf[0] {
 	case decimalNaN:
 		dec.Form = apd.NaN

--- a/pkg/util/encoding/decimal_test.go
+++ b/pkg/util/encoding/decimal_test.go
@@ -472,6 +472,46 @@ func TestNonsortingEncodeDecimalRoundtrip(t *testing.T) {
 	}
 }
 
+func TestDecodeMultipleDecimalsIntoNonsortingDecimal(t *testing.T) {
+	tcs := []struct {
+		value []string
+	}{
+		{
+			[]string{"1.0", "5.0", "7.0"},
+		},
+		{
+			[]string{"1.0", "-1.0", "0.0"},
+		},
+		{
+			[]string{"1.0", "-1.0", "10.0"},
+		},
+		{
+			[]string{"nan", "1.0", "-1.0"},
+		},
+		{
+			[]string{"-1.0", "inf", "5.0"},
+		},
+	}
+
+	for _, tc := range tcs {
+		var actual apd.Decimal
+		for _, num := range tc.value {
+			expected, _, err := apd.NewFromString(num)
+			if err != nil {
+				t.Fatal(err)
+			}
+			enc := EncodeNonsortingDecimal(nil, expected)
+			err = DecodeIntoNonsortingDecimal(&actual, enc, nil)
+			if err != nil {
+				t.Fatal(err)
+			}
+			if actual.Cmp(expected) != 0 {
+				t.Errorf("unexpected mismatch for %v, got %v", expected, &actual)
+			}
+		}
+	}
+}
+
 func TestUpperBoundNonsortingDecimalUnscaledSize(t *testing.T) {
 	x := make([]byte, 100)
 	d := new(apd.Decimal)

